lib/git.js

```diff
@@ -306,7 +306,8 @@ async function isBranchUpToDate(repositoryUrl, branch, execaOptions) {
  */
 async function getNote(ref, execaOptions) {
   try {
-    return JSON.parse((await execa('git', ['notes', '--ref', GIT_NOTE_REF, 'show', ref], execaOptions)).stdout);
+    const commit = (await execa('git', ['log', '-1', '--pretty=format:%H', ref], execaOptions)).stdout;
+    return JSON.parse((await execa('git', ['notes', '--ref', GIT_NOTE_REF, 'show', commit], execaOptions)).stdout);
```

Comment:
Hey @Xunnamius do you think you could add a test to prevent regressions in future? 

As mentioned in #1708 we appreciate your thorough research and when in doubt I'd rather merge your change and trust our test suite that no existing use case broke.

@travi brought up this:

> My only concern would be if it is something that considers details across branches. I think we're should keep things working from the history of a single branch.

Any thoughts?